app/router/schema.js

```diff
@@ -6,6 +6,18 @@ const {
 	deactivateSchema,
 	diffSchemas,
 } = require('../controller/schema');
+const { producer } = require('../kafka/producer');
+const config = require('../config');
+
+let kafkaProducer;
+
+if (config.asyncSchemaUpdates) {
+	producer()
+		.then((p) => {
+			kafkaProducer = p;
+		})
+		.catch(console.error);
+}
```

Comment:
I think producer initialization shouldn't be in router. Router is responsible for HTTP endpoint registration.
Maybe move it to `app/index.js` somewhere in init